### PR TITLE
Stop modifying local filesystem when running gcloud e2e test

### LINKED ISSUE #57: e2e tests are messing with global docker config.json
this is surprising as a user of e2e tests since changes to .docker/config.json affects my other operations on the computer. can we find an alternative way of testing whatever needs to be tested?

### PR DESCRIPTION
**Pull request by DennisDenuto**
Attempts to fix #57 

- Uses a temp directory to contain the docker config.json

Authored-by: Dennis Leon <leonde@vmware.com>

### CODE REVIEW

File: test/e2e/gcloud_expired_token_test.go [hunk 1]
```diff
@@ -45,49 +44,49 @@ images:
...
+					"gcr.io": "gcloud-race-condition-db-error"
+			}
+		}`), os.ModePerm)
 
-	err = ioutil.WriteFile(dockerConfigPath, dockerConfigJSONContents, os.ModePerm)
 	require.NoError(t, err)
 
-	// restore docker config
+	// Retrieve the docker image
+	exec.Command("docker", "pull", "ubuntu:21.04").Run()
```
Comment (cppforlife) [inline test/e2e/gcloud_expired_token_test.go:86]: what is the purpose of this command? also same for docker volume rm

Comment (DennisDenuto) [reply test/e2e/gcloud_expired_token_test.go:86]: It is to cache the image before the `gcloud-race-condition-db-error` plugin is called.

The `gcloud-race-condition-db-error` runs a docker command (using the ubuntu:21.04) image. If it isn't cached, then that plugin will download that image (which takes time), and the keychain will timeout/fail. (We want it to fail for a different reason)

Comment (DennisDenuto) [reply test/e2e/gcloud_expired_token_test.go:86]: `docker volume rm` is run during the cleanup phase of the test. It is to remove a docker volume created by the test.

Comment (cppforlife) [reply test/e2e/gcloud_expired_token_test.go:86]: @DennisDenuto i think we should add this as a comment on top of the docker pull command. no way somebody would get this context from reading that line.

Comment (DennisDenuto) [reply test/e2e/gcloud_expired_token_test.go:86]: done

**Pull request merged**